src/rez/utils/platform_.py

```diff
@@ -466,12 +466,6 @@ class WindowsPlatform(Platform):
     name = "windows"
 
     def _arch(self):
```

Comment:
We can remove this method entirely now. There is no point in overriding it from the parent.